pages/auth/signup.tsx

```diff
@@ -43,8 +46,15 @@ export default function SignUpPage() {
   });
 
   const onSubmit: SubmitHandler<LoginFormInputs> = (data) => {
-    const { passwordConfirm, ...emailAuthVariables } = data;
+    const { passwordConfirm, name, ...emailAuthVariables } = data;
     signup(emailAuthVariables);
+    console.log("ran signup");
+    if (currentUser) {
+      UserService.updateName(currentUser.id, name);
```

Comment:
you have either `await` the async `signup` method on line 50 or integrate the name attribute in the signup method

Reply:
at the moment `currentUser` will always be false

Reply:
it is giving an error when I add await because signup is not async. Should I add signup to be async?

Reply:
thats strange signup is an async method in `AuthContextProvider`

Reply:
maybe because it is not directly calling it in signup but going through useSignUp.tsx

Reply:
ohh your using react-query `useMutation` here, I would opt to use the signup in `AuthContextProvider`. However if you want you can use `mutateAsync` which `useMutation` returns, but you also got to make sure that you are properly updating the `AuthContext`